### ISSUE TITLE
Backward compatibility

### ISSUE THREAD

**Issue by grylem** [slewsys/ed#3]
Hi
macOS version o the ed must be posix compliant

printf %s\\n H '# test' q | ed
?
script, line 2: unknown command

your versions works like gnu ed

printf %s\\n H '# test' q | ed
no any messages

if add -G still works like gnu ed, is that how it should be? If yes, how to use your version of ed as posix compatible?

**Comment by slewsys** [comment]
Support for comments is not traditional in ed, so:
```shell
printf "%\n" H '# test' | ed -G
```
should really print an error. I'll correct that.  Thank you!

**Comment by slewsys** [comment]
By the way, macOS ed actually quits, which is incorrect.  When ed reads commands from a pipe, the traditional behavior is to reports errors, but not exit.  A lot of scripts depend on this behavior.  macOS ed is actually an early version of my code, from before this was corrected.  Unfortunately, there's no way to fix it now :(

**Comment by grylem** [comment]
Thanks for explanaion.

P.S.
in manpage and texinfo file info about options as `-eIRx` are missed. Please fix it.

**Comment by slewsys** [comment]
Yeah, the documentation needs a lot of work.  In the mean time, the file doc/SUSv4.md does a little better job of covering extensions.

HEAD of branch _without-gnulib_ should now be in sync with _main_ branch.

**Comment by slewsys** [comment]
If system provides `getopt_long`, `regcomp` and `regexec`, then ed v2.1.1 is not be linked against Gnulib (unless explicitly asked to). So branch *without-gnulib* can be retired.
